plugins/module_utils/network/sonic/config/route_maps/route_maps.py

```diff
@@ -0,0 +1,2073 @@
+            return []
+        return route_maps_facts
+
+    def execute_module(self):
+        """ Execute the module
+
+        :rtype: A dictionary
+        :returns: The result from module execution
+        """
+        result = {'changed': False}
+        warnings = list()
+
+        existing_route_maps_facts = self.get_route_maps_facts()
+        commands, requests = self.set_config(existing_route_maps_facts)
+        if commands and len(requests) > 0:
+            if not self._module.check_mode:
+                try:
+                    edit_config(self._module, to_request(self._module, requests))
+                except ConnectionError as exc:
+                    self._module.fail_json(msg=str(exc), code=exc.errno)
+            result['changed'] = True
+        result['commands'] = commands
+
+        changed_route_maps_facts = self.get_route_maps_facts()
+
+        result['before'] = existing_route_maps_facts
+        if result['changed']:
+            result['after'] = changed_route_maps_facts
+
+        result['warnings'] = warnings
+        return result
+
+    def set_config(self, existing_route_maps_facts):
+        """ Collect the configuration from the args passed to the module,
+            collect the current configuration (as a dict from facts)
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        want = remove_empties_from_list(self._module.params['config'])
+        have = existing_route_maps_facts
+        resp = self.set_state(want, have)
+        return to_list(resp)
+
+    def set_state(self, want, have):
+        """ Select the appropriate function based on the state provided
+
+        :param want: the desired configuration as a dictionary
+        :param have: the current configuration as a dictionary
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+        state = self._module.params['state']
+        if state == 'deleted':
+            commands, requests = self._state_deleted(want, have)
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+        elif state == 'overridden':
+            commands, requests = self._state_overridden(want, have)
+        elif state == 'replaced':
+            commands, requests = self._state_replaced(want, have)
+        return commands, requests
+
+    def _state_replaced(self, want, have):
+        """ The command generator when state is replaced
+
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        # Delete replaced groupings
+        commands = deepcopy(want)
```

Comment:
I see what you mean. In addition, "want" isn't used again in the code flow for handling "replaced" state. I have removed the "deepcopy" invocation.

Reply:
Well, it turns out that this instance is needed. The remove empties, etc. happens before this in order to be common for all cases, and the delete processing sections for "replaced" state and "overridden" state change the contents of "commands" (a passed copy of the "want" object of deepcopy is not used here) based on what is being deleted so that the correct commands can be reflected via "update states" for the delete phase.

If a deepcopy is not done here, the "real" content of "want" is lost during the "delete" phases for these two states. For this reason, I have added it back.

Reply:
My mistaken reason for thinking it was okay to delete the use of deepcopy here (left in place to provide context) does, however, apply to the "deleted" case and I am maintaining that removal of the deepcopy invocation.

Reply:
Thanks for the details.